pipeline2/capturing_user_input_data.py

```diff
@@ -1,103 +1,84 @@
-root.frame = 0
-screen_width = root.winfo_screenwidth()
-screen_height = root.winfo_screenheight()
-# Create full screen canvas
-root.state('zoomed')
-root.attributes('-fullscreen', True)
-canvas = tk.Canvas(root, bg='black', highlightthickness=0)
-canvas.pack(fill=tk.BOTH, expand=True)
-
-pts = points(n, screen_width, screen_height)
-points_iterator = iter(pts)
-
-cam = cv2.VideoCapture(0)
-now = get_timestamp()
-output = './train_data2/raw/{}'.format(now)
-os.makedirs(output+'/images')
-fw = open('{}/positions.csv'.format(output), 'w')
-fw.write("frame,timestamp,x,y,screen_width,screen_height,image_path,image_width,image_height\n")
-
-def close(e):
-    cam.release()
-    root.destroy()
-
-root.bind('<Escape>', close)
-
-def circle(x, y, radius, canvasName, color="yellow"):
-    x0 = x-radius
-    y0 = y-radius
-    x1 = x+radius
-    y1 = y+radius
-    return canvasName.create_oval(x0, y0, x1, y1, fill=color, outline=color, tags="circle")
-
-
-def draw(canvasName):
-    try:
-        p = next(points_iterator)
-        circle(p[0], p[1], radius, canvasName)
-        canvas.tag_bind("circle", "<1>", on_click)
-    except StopIteration:
-        close(StopIteration)
-
-
-def on_click(event):
-    root.frame += 1
-    timestamp = get_timestamp()
-    x, y = event.x, event.y
-    ret_val, img = cam.read()
-    img_height, img_width = img.shape[0:2]
-    image_path = "images/{}_{:03d}.jpg".format(timestamp, root.frame)
-    fw.write("{},{},{},{},{},{},{},{},{}\n".format(root.frame, timestamp,
-             x, y, screen_width, screen_height, image_path, img_width, img_height))
-    fw.flush()
-    cv2.imwrite("{}/{}".format(output, image_path), img)
-    canvas.delete("circle")
-    draw(canvas)
-
-
+from libs.utils import get_timestamp, points
+
+
+class Capture:
+    def __init__(self, master, n, output_path):
+        self.master = master
+        self.screen_width = self.master.winfo_screenwidth()
+        self.screen_height = self.master.winfo_screenheight()
+        self.master.state('zoomed')
+        self.master.attributes('-fullscreen', True)
+        self.master.bind('<Escape>', self.close)
+        self.canvas = tk.Canvas(master, bg='black', highlightthickness=0)
+        self.canvas.pack(fill=tk.BOTH, expand=True)
+        self.points = points(n, self.screen_width, self.screen_height)
+        self.counter = tk.IntVar(value=0)
+        self.counter.trace("w", self.update_point)
+        self.radius = 2.5
+        self.point_color = "yellow"
+        self.info_text = self.canvas.create_text(
+            self.screen_width / 2, 10, fill="white", font="Courier 10")
+        self.cam = cv2.VideoCapture(0)
+        self.output = '{}/raw/{}'.format(output_path, get_timestamp())
+        os.makedirs(self.output + '/images')
+        self.fw = open('{}/positions.csv'.format(self.output), 'w')
+        self.fw.write(
+            "frame,timestamp,x,y,screen_width,screen_height,image_path,image_width,image_height\n")
+        self.update_point()
```

Comment:
that should go into run() method to avoid doing real work in constructor. Constructor should only initialise object